core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -164,19 +164,25 @@ public Table loadTable(TableIdentifier ident) {
   @Override
   public boolean dropTable(TableIdentifier ident, boolean purge) {
     boolean dropped = catalog.dropTable(ident, purge);
-    invalidate(canonicalizeIdentifier(ident));
+    invalidate(ident);
     return dropped;
   }
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier to) {
     catalog.renameTable(from, to);
-    invalidate(canonicalizeIdentifier(from));
+    invalidate(from);
+  }
+
+  @Override
+  public void invalidateTable(TableIdentifier ident) {
```

Comment:
Minor: I don't think that we need both this and the private `invalidate` method, since this just calls the private one. We could just call `invalidateTable` everywhere instead.